packages/template-block/src/components/TemplatePreview.tsx

```diff
@@ -0,0 +1,173 @@
+    blockSettings: Settings;
+    template: Template | null;
+    onOpenTemplateChooser: () => void;
+};
+
+export const TemplatePreview = ({
+    appBridge,
+    blockSettings,
+    template,
+    onOpenTemplateChooser,
+}: TemplatePreviewProps) => {
+    const { blockAssets } = useBlockAssets(appBridge);
+    const isEditing = useEditorState(appBridge);
+
+    const {
+        hasBackgroundTemplatePreview,
+        backgroundColorTemplatePreview,
+        hasBorder_templatePreview,
+        borderWidth_templatePreview,
+        borderColor_templatePreview,
+        borderStyle_templatePreview,
+        hasRadius_templatePreview,
+        radiusValue_templatePreview,
+        radiusChoice_templatePreview,
+        textRatio,
+        preview,
+        isPreviewHeightCustom,
+        previewHeightSimple,
+        previewHeightCustom,
+        previewDisplay,
+        previewImageAnchoring,
+        textPositioning,
+    } = blockSettings;
+
+    const { previewCustom } = blockAssets;
+
+    const flexDirection = textPositioningToFlexDirection[textPositioning];
+    const isRows = flexDirection === 'row' || flexDirection === 'row-reverse';
+    const borderRadius = hasRadius_templatePreview
+        ? radiusValue_templatePreview
+        : radiusStyleMap[radiusChoice_templatePreview];
+    const border = hasBorder_templatePreview
+        ? `${borderWidth_templatePreview} ${borderStyle_templatePreview} ${toRgbaString(borderColor_templatePreview)}`
+        : 'none';
+    const height = isPreviewHeightCustom ? previewHeightCustom : previewHeightValues[previewHeightSimple];
+
+    const [isActionFlyoutOpen, setIsActionFlyoutOpen] = useState(false);
+
+    return (
+        <div
+            data-test-id="template-block-preview"
+            style={{
+                width: isRows ? `${100 - parseInt(textRatio)}%` : '100%',
+            }}
+        >
+            {template !== null || previewCustom ? (
+                <div
+                    className="tw-relative tw-overflow-hidden"
+                    data-test-id="template-block-preview-wrapper"
+                    style={{
+                        ...(hasBackgroundTemplatePreview && {
+                            ...getBackgroundColorStyles(backgroundColorTemplatePreview),
+                        }),
+                        borderRadius,
+                        border,
+                        height,
+                    }}
+                >
+                    <img
+                        data-test-id="template-block-preview-img"
+                        src={
+                            preview === PreviewType.Custom && previewCustom
+                                ? previewCustom[0].previewUrl
+                                : template?.previewUrl
+                        }
+                        className={merge([
+                            'tw-relative tw-w-full tw-h-full',
+                            `tw-object-${previewDisplayValues[previewDisplay]}`,
```

Comment:
That doesn't work: [Docs](https://v2.tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support) => `Dynamic values`, please create a map

Reply:
Fixed